Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp

```diff
@@ -2096,46 +2109,48 @@ inline void OMGIRGenerator::emitWriteBarrier(Value* cell, Value* instanceCell)
 
 inline Value* OMGIRGenerator::emitCheckAndPreparePointer(Value* pointer, uint32_t offset, uint32_t sizeOfOperation)
 {
-#if !OMG_JSVALUE_32_64_CAN_HANDLE_MEMORY
-    UNUSED_PARAM(pointer);
-    UNUSED_PARAM(offset);
-    UNUSED_PARAM(sizeOfOperation);
-    RELEASE_ASSERT_NOT_REACHED();
-#else
-    static_assert(GPRInfo::wasmBaseMemoryPointer != InvalidGPRReg);
-
     switch (m_mode) {
     case MemoryMode::BoundsChecking: {
         // We're not using signal handling only when the memory is not shared.
         // Regardless of signaling, we must check that no memory access exceeds the current memory size.
-        static_assert(GPRInfo::wasmBoundsCheckingSizeRegister != InvalidGPRReg);
         ASSERT(sizeOfOperation + offset > offset);
-        m_currentBlock->appendNew<WasmBoundsCheckValue>(m_proc, origin(), GPRInfo::wasmBoundsCheckingSizeRegister, pointer, sizeOfOperation + offset - 1);
+        Value* pointerPlusOffset;
```

Comment:
Why can't we just lower the WasmBoundsCheckValue?

Reply:
We can, though we'd have to extend WasmBoundsCheckValue to seed it with the Instance address, right? This was not a conscious decision, this part comes from the pre-B3LowerInt64 version, but ATM looks more self-contained to me. If you think leaving the `WasmBoundsCheckValue` and lowering later is less of a layering violation, I'm happy to do that instead.

Reply:
Merging this now after an OK in private.